### PR TITLE
Task/cbl 618

### PR DESCRIPTION
**Pull request by bmeike**
Odds and ends: Some things that got left out of CBL-541 and some cleanup during memory analysis.

### CODE REVIEW

File: lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLDict.java [hunk 1]
```diff
@@ -56,25 +56,22 @@ public FLValue get(String key) {
...
-        finally {
-            itr.free();
+
+        itr.begin(this);
+        String key;
+        while ((key = itr.getKeyString()) != null) {
+            results.put(key, itr.getValue().asObject());
+            itr.next();
         }
+
```
Comment (Sandychuang8) [inline lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLDict.java]: Don't you need to free the FLDictIterator after usage?

File: lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLArray.java [hunk 2]
```diff
@@ -20,72 +20,69 @@
...
+        while ((value = itr.getValue()) != null) {
+            results.add((T) value.asObject());
+            if (!itr.next()) { break; }
         }
-        return results;
-    }
 
-    public long count() {
-        return count(handle);
+        return results;
```
Comment (Sandychuang8) [inline lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLArray.java]: Don't you need to free FLArrayIterator after usage?

File: lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLDict.java (hunk 1 above)
Comment (bmeike) [reply lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLDict.java]: Well... its finalizer should free it, when it is GCed...

File: lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLArray.java (hunk 2 above)
Comment (bmeike) [reply lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLArray.java]: Same deal.  Its finalizer should take care of that...

**Pull request merged**